### PR TITLE
correction for issues #20 and #21

### PR DESCRIPTION
**Pull request by balintTobik**
Automatically adjust the cpuset of the pause containers to the default pool

### CODE REVIEW

File: pkg/sethandler/controller.go [hunk 1]
```diff
@@ -233,3 +246,50 @@ func (setHandler *SetHandler) applyCpusetToContainer(containerID string, cpuset
 	}
 	return nil
 }
+
+func (setHandler *SetHandler) applyCpusetToInfraContainer(pod v1.Pod) error{
+	cpuset := setHandler.poolConfig.SelectPool(types.DefaultPoolID).CPUs
+	if cpuset.IsEmpty() {
+		//Nothing to set. We will leave the container running on the Kubernetes provisioned default cpuset
+		log.Println("WARNING: for some reason DEFAULT cpuset to set was quite empty. I left it untouched.")
```
Comment (Levovar) [inline pkg/sethandler/controller.go]: left what untouched? this error message is not helpful in run-time

File: pkg/sethandler/controller.go [hunk 2]
```diff
@@ -233,3 +246,50 @@ func (setHandler *SetHandler) applyCpusetToContainer(containerID string, cpuset
...
+		return nil
+	}
+	containerID := pod.Status.ContainerStatuses[0].ContainerID
+	trimmedCid := strings.TrimPrefix(containerID, "docker://")
+	var pathToSearchContainer, pathToContainerCpusetFile, infraContainerID string
```
Comment (Levovar) [inline pkg/sethandler/controller.go]: this looks somewhat copy-pasted. you call this function already after we have identified the cpuset root directory
you could simply return it , or save it during the first iteration, and simply pass it onto this function

File: pkg/sethandler/controller.go [hunk 3]
```diff
@@ -233,3 +246,50 @@ func (setHandler *SetHandler) applyCpusetToContainer(containerID string, cpuset
...
+		if strings.HasSuffix(path, trimmedCid) {
+			pathToSearchContainer = path
+		}
+		return nil
+	})
+	if pathToSearchContainer == "" {
+		return errors.New("container directory " + trimmedCid + " does not exists under the provided cgroupfs hierarchy:" + setHandler.cpusetRoot)
+	}
+
+	filelist, _ := filepath.Glob(filepath.Dir(pathToSearchContainer) + "/*")
```
Comment (Levovar) [inline pkg/sethandler/controller.go]: this could be a function, returning the name of the infra container

**Comment by balintTobik** [comment]
About kubepods mount, I have tested your idea, and it seemed to be works fine.

**Pull request merged**